Client/U2FExtensions.swift

```diff
@@ -0,0 +1,893 @@
+            return
+        }
+    }
+    
+    private func finalizeFIDO2Authentication(handle: Int, response: YKFKeyFIDO2GetAssertionResponse, requestId: String, clientDataJSON: Data, error: NSErrorPointer) {
+        guard error == nil else {
+            let errorDescription = error?.pointee?.localizedDescription ?? Strings.U2FAuthenticationError
+            sendFIDO2AuthenticationError(handle: handle, errorDescription: errorDescription)
+            return
+        }
+        
+        let authenticatorData = response.authData.base64EncodedString()
+        let clientDataJSONString = clientDataJSON.base64EncodedString()
+        let sig = response.signature.base64EncodedString()
+        
+        cleanupFIDO2Authentication(handle: handle)
+        ensureMainThread {
+            self.tab?.webView?.evaluateJavaScript("navigator.credentials.postGet('\(handle)', \(true), '\(requestId)', '\(authenticatorData)', '\(clientDataJSONString)', '\(sig)', '')", completionHandler: { _, error in
+                if error != nil {
+                    let errorDescription = error?.localizedDescription ?? U2FErrorMessages.ErrorAuthentication.rawValue
+                    log.error(errorDescription)
+                }
+            return
+        }) }
+    }
+    
+    private func sendFIDO2AuthenticationError(handle: Int, errorName: String = FIDO2ErrorMessages.NotAllowedError.rawValue, errorDescription: String = Strings.U2FAuthenticationError) {
+        cleanupFIDO2Authentication(handle: handle)
+        ensureMainThread {
+            self.tab?.webView?.evaluateJavaScript("navigator.credentials.postGet('\(handle)', \(true), '', '', '', '', '\(errorName)', '\(errorDescription)')", completionHandler: { _, error in
+                if error != nil {
+                    let errorDescription = error?.localizedDescription ?? U2FErrorMessages.ErrorAuthentication.rawValue
+                    log.error(errorDescription)
+                }
+            return
+        }) }
+    }
+    
+    private func cleanupFIDO2Authentication(handle: Int) {
+        guard let index = fido2AuthHandles.firstIndex(of: handle) else {
+            log.error(U2FErrorMessages.ErrorRegistration)
+            return
+        }
+        fido2AuthHandles.remove(at: index)
+        fido2AuthRequest.removeValue(forKey: handle)
+    }
+    
+    // FIDO Registration
+    private func requestFIDORegistration(handle: Int, requests: [FIDORegisterRequest], id: Int) {
+        fidoRegHandles.append(handle)
+        
+        guard let request = requests.first else {
+            sendFIDORegistrationError(handle: handle, requestId: id, errorCode: U2FErrorCodes.bad_request)
+            return
+        }
+    
+        fidoRegisterRequest[handle] = request
+        
+        if id > -1 {
+            requestId[handle] = id
+        }
+
+        guard YubiKitManager.shared.keySession.sessionState == .open else {
```

Comment:
In general, `guard` is not used when no value is unwrapped.
It's better to use `if YubiKitManager.shared.keySession.sessionState != .open { return }`